plugins/signature_provider_plugin/signature_provider_plugin.cpp

```diff
@@ -42,6 +42,11 @@ class signature_provider_plugin_impl {
       std::tuple<std::string, std::string, std::string> parse_spec(const std::string& spec) const {
          auto delim = spec.find("=");
          EOS_ASSERT(delim != std::string::npos, chain::plugin_config_exception, "Missing \"=\" in the key spec pair");
+         // public_key can be base64 encoded with trailing `=`
+         // e.g. --signature-provider PUB_BLS_Fmgk<snip>iuA===KEY:PVT_BLS_NZhJ<snip>ZHFu
+         while( spec.size() > delim+1 && spec[delim+1] == '=' )
+            ++delim;
+         EOS_ASSERT(delim < spec.size() + 1, chain::plugin_config_exception, "Missing spec data in the key spec pair");
          auto pub_key_str = spec.substr(0, delim);
```

Comment:
won't `pub_key_str` potentially end with multiple `=` characters? Maybe it is not an issue?

Reply:
Yes, which like the example comment shows is rather ugly. We will likely move to base64url in the future partly do to this.

Reply:
So should we do something like this:
```
         auto delim_end = delim;
         while( spec.size() > delim_end+1 && spec[delim_end+1] == '=' )
            ++delim_end;
         EOS_ASSERT(delim_end < spec.size() + 1, chain::plugin_config_exception, "Missing spec data in the key spec pair");
         auto pub_key_str = spec.substr(0, delim);
         auto spec_str = spec.substr(delim_end + 1);
```

Reply:
No, The public key is `PUB_BLS_Fmgk<snip>iuA==` the separator is `=KEY:`. Also note the separator can also be `=KEOSD:` and `=SE:`.

Reply:
Oh, OK. No spaces possible, right?

Reply:
Correct, no spaces allowed.